src/main/java/com/coveros/selenified/utilities/Property.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright 2019 Coveros, Inc.
+ *
+ * This file is part of Selenified.
+ *
+ * Selenified is licensed under the Apache License, Version
+ * 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy
+ * of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.coveros.selenified.utilities;
+
+import com.coveros.selenified.Browser;
+import com.coveros.selenified.exceptions.InvalidHTTPException;
+import org.testng.ITestContext;
+import org.testng.log4testng.Logger;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Properties;
+
+/**
+ * Reads in properties files provided by the user in order to execute tests. These
+ * files can be passed in via commandline, or set in a selenified.properties file.
+ * This file should reside in src/test/resources. If the property exists in the
+ * system properties, that is returned, overridding anything in the
+ * selenified.properties file.
+ *
+ * @author Max Saperstone
+ * @version 3.2.0
+ * @lastupdate 3/22/2019
+ */
+public class Property {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1118)